### PR TITLE
Update to RC1

### PR DESCRIPTION
**Pull request by ktoso**
It just works :-)
Will use FileSource in next release - needs built in lines suport.

### CODE REVIEW

**Comment by patriknw** [comment]
is this in sync with the scala template?

File: src/main/java/sample/stream/TcpEcho.java [hunk 1]
```diff
@@ -61,14 +57,13 @@ public static void main(String[] args) {
...
+    final Sink<IncomingConnection, Future<BoxedUnit>> handler = Sink.runForeach(conn -> {
       System.out.println("Client connected from: " + conn.remoteAddress());
       conn.handleWith(Flow.<ByteString>empty(), materializer);
     });
 
-    final Future<ServerBinding> bindingFuture = 
-        StreamTcp.get(system)
-          .bind(serverAddress).to(handler).run(materializer);
+    final Future<ServerBinding> bindingFuture =
+      Tcp.get(system).bind(serverAddress.getHostString(), serverAddress.getPort()).to(handler).run(materializer);
```
Comment (patriknw) [inline src/main/java/sample/stream/TcpEcho.java]: getHostString this requires java7, I guess that is fine, but we could also skip creating the InetSocketAddress in the first place

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
LGTM, but we might want to do a sync effort of this and the scala template afterwards

**Comment by ktoso** [comment]
This will need more love to use file sink and source more nicely, but due to the just now discovered problem I dropped that part. I'll open an issue to not forget about it

**Pull request merged**